frontendProviderServices/src/main/java/com/virtuslab/gitmachete/frontend/ui/providerservice/VcsRootComboBoxProvider.java

```diff
@@ -0,0 +1,28 @@
+package com.virtuslab.gitmachete.frontend.ui.providerservice;
+
+import com.intellij.openapi.components.Service;
+import com.intellij.openapi.project.Project;
+import git4idea.repo.GitRepository;
+import io.vavr.control.Option;
+import lombok.Getter;
+
+import com.virtuslab.binding.RuntimeBinding;
+import com.virtuslab.gitmachete.frontend.ui.api.vcsrootcombobox.BaseVcsRootComboBox;
+import com.virtuslab.gitmachete.frontend.ui.api.vcsrootcombobox.IVcsRootComboBoxFactory;
+
+@Service
+public final class VcsRootComboBoxProvider {
```

Comment:
Hmm so that we could stop exposing anything about `VcsRootComboBox` from `frontendProviderService` subproject... how about:
1) removing the getter for `vcsRootComboBox`
2) just exposing `getSelectedVcsRepository` and `getVcsRepositorySelectionComponent` (of e.g. `JComponent` type)?

Reply:
And 3. renaming the classes/packages accordingly so that the details about any `ComboBox`es don't leak outside of the impl module where it's defined

Reply:
done